src/main/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjector.java

```diff
@@ -63,12 +65,45 @@ public static DocumentProjector createFromDefinition(
     return PathCollector.collectPaths(projectionDefinition, includeSimilarity).buildProjector();
   }
 
+  public static DocumentProjector createForIndexing(Set<String> allowed, Set<String> denied) {
+    // Sets are expected to be validated to have one of 3 main cases:
+    // 1. Non-empty "allowed" (but empty/null "denied") -> build inclusion projection
+    // 2. Non-empty "denied" (but empty/null "allowed") -> build exclusion projection
+    // 3. Empty/null "allowed" and "denied" -> return identity projection
+    // as well as 2 special cases:
+    // 4. Empty "allowed" and single "*" entry for "denied" -> return exclude-all projection
+    // 5. Empty "deny" and single "*" entry for "allowed" -> return include-all ("identity")
+    // projection
+    // We need not (and should not) do further validation here.
+    // Note that (5) is effectively same as (3) and included for sake of uniformity
+    if (allowed != null && !allowed.isEmpty()) {
+      // (special) Case 5:
+      if (allowed.size() == 1 && allowed.contains("*")) {
```

Comment:
allowed can't have `*`, it's only supported for denied.

Reply:
That is something I wanted to discuss (should have brought up) -- for consistency it would make sense to allow this. But for minimal approach not. Will ask on Stargate channel.

My main concern is if users would pass this, assuming it works, and it meaning something totally different.